duckpgq/src/duckpgq/functions/scalar/shortest_path.cpp

```diff
@@ -139,91 +182,59 @@ static void ShortestPathFunction(DataChunk &args, ExpressionState &state,
     //! make passes while a lane is still active
     for (int64_t iter = 1; active; iter++) {
       //! Perform one step of bfs exploration
-      if (!IterativeLength(v_size, v, e, edge_ids, parents_v, parents_e, seen,
+      if (!IterativeLength(v_size, v, e, edge_ids, parents_v, paths_v, paths_e, seen,
                            (iter & 1) ? visit1 : visit2,
                            (iter & 1) ? visit2 : visit1)) {
         break;
       }
-      int64_t finished_searches = 0;
       // detect lanes that finished
       for (int64_t lane = 0; lane < LANE_LIMIT; lane++) {
         int64_t search_num = lane_to_num[lane];
         if (search_num >= 0) { // active lane
           //! Check if dst for a source has been seen
           int64_t dst_pos = vdata_dst.sel->get_index(search_num);
           if (seen[dst_data[dst_pos]][lane]) {
-            finished_searches++;
+            // check if the path length is within bounds
+            // bound vector is either a constant or a flat vector
+            if (lower_bound.GetVectorType() == VectorType::CONSTANT_VECTOR ? 
```

Comment:
Wouldn't it be easier to extract the `lower_bound_data` before starting the searches? It will be the same for every tuple in the input so you don't have to check for every tuple.

Reply:
For formal syntax like `ANY SHORTEST (a:Point4)-[k:know4]->{2,3}(b:Point4)`, this is true. But for expanded function calls, the lower and upper bounds can be variables, and so each pair may have different upper and lower bounds. Since users should only use formal syntax, I would remove this code.